doc/source/ray-core/runtime_env_auth.md

```diff
@@ -0,0 +1,120 @@
+(runtime-env-auth)=
+# Authenticating Remote URIs in runtime_env
+
+This section helps you:
+
+* Avoid leaking remote URI credentials in your `runtime_env`
+* Provide credentials safely in KubeRay
+* Understand best practices for authenticating your remote URI
+
+## Authenticating Remote URIs
+
+You can add dependencies to your `runtime_env` with [remote URIs](remote-uris). This is straightforward for files hosted publicly, because you simply paste the public URI into your `runtime_env`:
+
+```python
+runtime_env = {"working_dir": (
+        "https://github.com/"
+        "username/repo/archive/refs/heads/master.zip"
+    )
+}
+```
+
+However, dependencies hosted privately, in a private GitHub repo for example, require authentication. One common way to authenticate is to insert credentials into the URI itself:
+
+```python
+runtime_env = {"working_dir": (
+        "https://username:personal_access_token@github.com/"
+        "username/repo/archive/refs/heads/master.zip"
+    )
+}
+```
+
+In this example, `personal_access_token` is a secret credential that authenticates this URI. While Ray can successfully access your dependencies using authenticated URIs, **you should not include secret credentials in your URIs** for two reasons:
+
+1. Ray may log the URIs used in your `runtime_env`, which means the Ray logs could contain your credentials.
+2. Ray stores your remote dependency package in a local directory, and it uses a parsed version of the remote URI–including your credential–as the directory's name.
+
+In short, your remote URI is not treated as a secret, so it should not contain secret info. Instead, use a `netrc` file.
+
+## Running on VMs: the netrc File
+
+The [netrc file](https://www.gnu.org/software/inetutils/manual/html_node/The-_002enetrc-file.html) contains credentials that Ray uses to automatically log into remote servers. Set your credentials in this file instead of in the remote URI:
+
+```bash
```

Comment:
nit: can we add another example for setting s3 credentials, this is other common use case

Reply:
Sure, we can do that. Since this PR is reviewed and nearly ready to merge, let's add that in a follow up. Do you have an example of how people are currently setting S3 credentials?

Reply:
I actually don't know. But people have feature request for this: #34708

Reply:
I see– I took a look at the request, and I don't think a `netrc` file solves the issue. There doesn't seem to be a way to provide an `endpoint_url` or the other S3-specific credentials requested in that issue through the `netrc` file. Let's continue to track that request in that issue.

Reply:
oh, that's a bummer. Ideally we should provide a unified way for authentication.